packages/react/direction/src/Direction.tsx

```diff
@@ -0,0 +1,33 @@
+import * as React from 'react';
+
+type Direction = 'ltr' | 'rtl';
+const DirectionContext = React.createContext<Direction | undefined>(undefined);
+
+/* -------------------------------------------------------------------------------------------------
+ * Direction
+ * -----------------------------------------------------------------------------------------------*/
+
+interface DirectionProviderProps {
+  dir: Direction;
+}
+const DirectionProvider: React.FC<DirectionProviderProps> = (props) => {
+  const { dir, children } = props;
+  return <DirectionContext.Provider value={dir}>{children}</DirectionContext.Provider>;
+};
+
+/* -----------------------------------------------------------------------------------------------*/
+
+function useDirection(localDir?: Direction) {
+  const globalDir = React.useContext(DirectionContext);
+  return localDir || globalDir || 'ltr';
```

Comment:
I moved the fallback logic into the hook as I found it clearer to see the cascade preference.

Reply:
That's funny cause Jenna had done that originally which I asked about: https://github.com/radix-ui/primitives/pull/1119#discussion_r792541890

> @benoitgrelard re your last round of comments – I couldn't see any reason not to provide the default ltr value so I resolved them and went for this, let me know what you think though.
Edit – I think chromatic has just shown me the reason 😅

Can you provide more detail about this bit, i'm not following.

Reply:
We checked in offline about this, the primary reason that I went back to passing an arg to the hook is for style. Because we can’t conditionally call hooks it’d be more verbose when done inline:

```tsx
const contextDir = useDirection();
const direction = dirContext || dirProp || 'ltr' ;
```

vs  

```tsx
const direction = useDirection(dirProp)
```

The latter of which we seem to prefer on balance.

> Can you provide more detail about this bit, i'm not following.

Seems that the original changes in this PR removed default values in certain places due to how some primitives get recomposed and the cascading effect of always setting a `dir` attribute from a default value. This is made clear by a preexisting bug in `Toolbar` where `ToolbarToggleGroup` doesn't respect `dir="rtl"` when added to `Root`, I think the changes were to remedy these issues.

We decided it would be clearer overall to provide a default in all cases and ensure we consistency pass `dir` values down the tree where necessary.